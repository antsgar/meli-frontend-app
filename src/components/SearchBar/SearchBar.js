import React, { useState } from "react";
import { useRouter } from "next/router";
import styles from "./SearchBar.module.scss";
import { BASE_URL } from "../../constants";

const SearchBar = ({ search }) => {
    const [searchQuery, setSearchQuery] = useState(search);
    const router = useRouter();

    const goToMainPage = () => {
        router.push("/");
    };

    const submitSearchQuery = (event) => {
        event.preventDefault();
        router.push("/items", `/items?search=${searchQuery}`);
    };

    return (
        <div className={styles.container} data-test="search-bar">
            <a className={styles.logoContainer} onClick={goToMainPage}>
                <img className={styles.logo} data-test="logo" src={"/Logo_ML@2x.png.png"} alt="Logo"></img>
            </a>
            <form className={styles.bar} onSubmit={submitSearchQuery}>
                <input
                    className={styles.searchBox}
                    data-test="search-box"
                    type="text"
                    placeholder="Nunca dejes de buscar"
                    value={searchQuery}
                    onChange={(e) => setSearchQuery(e.target.value)}
                ></input>
                <a className={styles.searchIconContainer} data-test="search-icon" onClick={submitSearchQuery}>
                    <img className={styles.searchIcon} src={"/ic_Search@2x.png.png"} alt="Logo"></img>
                </a>
            </form>
        </div>
    );
};

export default SearchBar;
