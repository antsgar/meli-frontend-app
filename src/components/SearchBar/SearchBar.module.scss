@import "../../styles/mixins";
@import "../../styles/vars";

.container {
    @include gridWrapper(12);

    background-color: $background-color-contrast;
    padding-top: $spacing-small;
    padding-bottom: $spacing-small;
    align-items: center;

    @include desktop {
        padding-left: $spacing-large;
        padding-right: $spacing-large;
    }
}

.logoContainer {
    grid-column: 2;
}

.logo {
    height: 40px;

    &:hover {
        cursor: pointer;
    }
}

.bar {
    @include gridContent(3, 12);
    height: 40px;
    display: flex;
    flex-direction: row;
}

.label {
    display: none;
}

.searchBox {
    flex-grow: 1;
    border: none;
    padding-left: $spacing-medium;
    color: $text-color-dark;
    font-size: 18px;
    border-radius: 4px 0 0 4px;
    margin: 0;

    ::placeholder {
        color: $text-color-light;
    }
}

.searchIconContainer {
    background-color: $background-color-default;
    padding: 10px;
    display: flex;
    border-radius: 0 4px 4px 0;

    &:hover {
        background-color: $hover-color-grey;
        cursor: pointer;
    }
}

.searchIcon {
    width: 20px;
}
