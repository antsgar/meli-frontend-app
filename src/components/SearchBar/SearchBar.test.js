import React from "react";
import { mount } from "enzyme";
import SearchBar from "./SearchBar";

const search = "iPhone";
let wrapper;

beforeEach(() => {
    wrapper = mount(<SearchBar search={search} />);
});

describe("SearchBar", () => {
    it("should render component", () => {
        expect(wrapper).toMatchSnapshot();
    });

    it("should display logo", () => {
        expect(wrapper.find(".logo").exists()).toEqual(true);
    });

    it("should show search query", () => {
        expect(wrapper.find(".searchBox").getElement().props.value).toEqual(
            search
        );
    });
});
