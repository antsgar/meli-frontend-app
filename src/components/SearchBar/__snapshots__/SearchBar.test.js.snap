// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should render component 1`] = `
<SearchBar
  search="iPhone"
>
  <div
    className="container"
    data-test="search-bar"
  >
    <a
      className="logoContainer"
      onClick={[Function]}
    >
      <img
        alt="Logo"
        className="logo"
        data-test="logo"
        src="/Logo_ML@2x.png.png"
      />
    </a>
    <form
      className="bar"
      onSubmit={[Function]}
    >
      <input
        className="searchBox"
        data-test="search-box"
        onChange={[Function]}
        placeholder="Nunca dejes de buscar"
        type="text"
        value="iPhone"
      />
      <a
        className="searchIconContainer"
        data-test="search-icon"
        onClick={[Function]}
      >
        <img
          alt="Logo"
          className="searchIcon"
          src="/ic_Search@2x.png.png"
        />
      </a>
    </form>
  </div>
</SearchBar>
`;
